src/main/webapp/app/entities/build-log/build-log.model.ts

```diff
@@ -15,6 +22,23 @@ type ParsedLogEntry = [string, string, string, string, string, string];
 export class BuildLogEntryArray extends Array<BuildLogEntry> {
     private errorLogRegex = /\[(ERROR)\].*\/(src\/.+):\[(\d+),(\d+)\]\s(.*$)/;
 
+    /**
+     * Factory method for creating an instance of the class. Prefer this method over the default constructor.
+     *
+     * @param buildLogs BuildLogEntry[]
+     */
+    static fromBuildLogs(buildLogs: BuildLogEntry[]) {
+        const mappedLogs = buildLogs.map(({ log, ...rest }) => {
+            const logType = log && log.trimLeft().startsWith('[ERROR]') ? BuildLogType.ERROR : log.trimLeft().startsWith('WARNING') ? BuildLogType.WARNING : BuildLogType.OTHER;
```

Comment:
I think we should better use `contains` and do it case insensitive. Otherwise we might miss some of the errors and warnings